### PR TITLE
feat: initial workshop creation

### PR DESCRIPTION
**Pull request by xbcsmith**

### CODE REVIEW

**Review by caproven — COMMENTED** [review]
Probably want to remove your chatgpt prompts before contents of this repo are shared with workshop participants (assuming they will be shared)

File: 01-intro/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+interspersed to address participant queries and foster engagement.
+
+## Key Learning Objectives
+
+Gain a foundational understanding of microservice architectures and their
+relevance in achieving scalability and modularity. Explore the benefits of
+event-driven asynchronous communication in CI/CD pipelines. Familiarize
+participants with the Event Provenance Registry (EPR) project and its core
+components. Understand the significance of Cloud Events and the Continuous
+Delivery Events (CDEvents) specification in standardizing event messages.
```
[the remark this answers is not in the document]
Comment (xbcsmith) [reply 01-intro/README.md:50]: Example?

File: 03-integration/01-redpanda.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+Navigate to the directory where you stored the file in a command prompt and
+deploy Redpanda using Docker Compose.
+
+```bash
+docker compose -f compose/docker-compose.yaml up
+```
+
+This command runs a single Redpanda broker, configured without authentication,
+accessible from `localhost:9092`. You can access the web console in your browser
+at [http://localhost:8080/overview](http://localhost:8080/overview).
```
[the remark this answers is not in the document]
Comment (xbcsmith) [reply 03-integration/01-redpanda.md:96]: I have a broker and web ui

File: 01-intro/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (caproven) [reply 01-intro/README.md:50]: I just mean formatting as
```md
- Gain a foundational understanding of microservice architectures and their
relevance in achieving scalability and modularity
- Explore the benefits of
event-driven asynchronous communication in CI/CD pipelines
- etc
```
so it's easier to read

[the remark this answers is not in the document]
Comment (xbcsmith) [reply 01-intro/README.md:50]: Got it

File: 08-epr-watcher/01-watcher.md [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+func customTaskHandler(msg *message.Message) error {
+ log.Default().Printf("I received a task with value '%v'", msg)
+ return nil
+}
+
+```
+
+Save the file and run `go mod init` then
+
+Now we can run `go mod tidy` to fill in our dependencies.
```
Comment (caproven) [inline 08-epr-watcher/01-watcher.md:70]: First sentence here trails off. Suggest removing "then" and just add a period

File: 08-epr-watcher/01-watcher.md [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+```
+
+**Note**: the matcher being run is looking for kafka messages with the value
+`match`. All other messages will be ignored.
```
Comment (caproven) [inline 08-epr-watcher/01-watcher.md:137]: Not sure I follow, it's matching on `"type": "foo.bar"` so I'd mention that. Possibly mention how this matches the receiver type that we created an event for.

```md
**Note**: the matcher being run is looking for kafka messages with the type
`foo.bar`. This aligns with the type of the event receiver we created and posted
an event to. All other messages will be ignored.
```

File: 09-epr-python/01-cli.md [hunk 5]
```diff
@@ -18,18 +18,30 @@ Search for an event receiver using the provided parameters:
...
+
+```bash
+eprcli create event-receiver --name bar-python-cli --version 1.0.0 --type dev.bar.python.cli --description "bar python cli event receiver" --schema "{}"
+```
+
+Create an event receiver group using the provided parameters:
+
+```bash
+eprcli create event-receiver-group --name foo-bar-python-cli --version 1.0.0 --type dev.foo.bar.python.cli --description "foo bar python cli event receiver group" --event-receiver-ids 01HWN885Z4D680AAASH81G1KXF,01HWN8NB7F13NMNSJ1SFK831KZ
+```
```
Comment (caproven) [inline 09-epr-python/01-cli.md:31]: Should mention these IDs are dynamic and come from output of previous commands

File: 09-epr-python/01-cli.md [hunk 6]
```diff
@@ -18,18 +18,30 @@ Search for an event receiver using the provided parameters:
...
+
 Create an event using the provided parameters:
 
 ```bash
-eprcli create event --name foo --version 1.0.1 --release 2023.11.16 --description "The Foo of Brixton" --payload '{"name": "foo"}' --success true --event-receiver-id 01HW3SZ8N3MXA9EWZZY4HSVVNK
+eprcli create event --name foo --version 1.0.1 --release 2023.11.16 --platform-id x86_64-gnu-linux-40 --package rpm  --description "The Foo of Brixton" --payload '{"name": "foo"}' --success --event-receiver-id 01HW3SZ8N3MXA9EWZZY4HSVVNK
```
Comment (caproven) [inline 09-epr-python/01-cli.md:36]: Mention that ID is dynamic

File: 10-epr-enhanced/00-intro.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+- Add a watcher to create a SBOM.
+
+## Python
+
+Add Python to the EPR system and expand on the example from the previous
+section.
+
+Suggestions:
+
+- Add a Python watcher to create an Event.
```
Comment (caproven) [inline 10-epr-enhanced/00-intro.md:42]: Is this not redundant with the previous "Watcher" entry? Should one be in Go and another in Python?

File: 01-intro/00-basics.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+on this waiting area. They're not just any cooks; they're experts in their
+craft. Only events that match their cooking expertise (or predefined criteria)
+make it to their attention. They don't waste time on things that aren't their
+specialty.
+
+But here's the cool part: each watcher has a unique recipe they're passionate
+about. They aren't interested in everything happening in the kitchen, just the
+events that align with their culinary skills. These events reach the watchers,
+and guess what they do? They swing into action, creating the magic in the
+kitchen – cooking up the perfect dish!
```
Comment (ninjamast3r) [inline 01-intro/00-basics.md:23]: This is a little redundant. I would combine these two paragraphs. Out of curiosity, was this section AI generated? I'm getting the vibes. It feels more excited than what I normally see from you.

File: 01-intro/00-basics.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+Microservices constitute a fundamental element within the Event Driven
+Architecture. We adhere to the Unix philosophy, "do one thing and do it well".
+This approach allows us to emphasize the principle of performing one task
+exceptionally well. Concentrating services on specific functions enables
+optimization and enhances reliability. Additionally, this approach facilitates
+scaling only the services that require it, avoiding the need to scale every
+service uniformly.
+
+## You want microservices this is how you get microservices?
```
Comment (ninjamast3r) [inline 01-intro/00-basics.md:47]:
```suggestion
## You want microservices? This is how you get microservices.
```

File: 01-intro/00-basics.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+but finding the right level of complexity is key. Simply increasing the number
+of microservices does not necessarily enhance maintainability or scalability.
+
+While microservices offer benefits like scalability and modularity, they also
+introduce logistical challenges. Backend services typically expose versioned
+APIs, requiring coordination among services. Additionally, this approach can
+lead to increased network traffic, potentially causing bottlenecks if not
+managed properly.
+
+## Why Event Driven Microservice Architecture?
```
Comment (ninjamast3r) [inline 01-intro/00-basics.md:62]: I'm seeing a mix of capitalizing every word in a heading vs capitalizing just the first. I would be consistent, at least in individual documents.

File: 01-intro/03-cdevents.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+
+### Event Message Design Pattern
+
+CDEvents leverages the Event Message design pattern to broadcast information
+related to activities within a Continuous Delivery process. This information is
+represented in a Normalized Form, ensuring consistency and clarity.
+
+### Broadcasting with Normalized Form
+
+Events in CDEvents are broadcasted, allowing information about Continuous
```
Comment (ninjamast3r) [inline 01-intro/03-cdevents.md:25]: nit: past tense is more typically just "broadcast", though both are correct.
```suggestion
Events in CDEvents are broadcast, allowing information about Continuous
```

**Review by ninjamast3r — CHANGES_REQUESTED** [review]
Didn't have time to get through the whole thing. Will try and follow up later.

**Review by ninjamast3r — CHANGES_REQUESTED** [review]
Got further. There's a lot here.

File: 10-epr-enhanced/00-intro.md (hunk 7 above)
Comment (xbcsmith) [reply 10-epr-enhanced/00-intro.md:42]: The idea with this section is to have participants use knowledge from previous sessions to do something on their own.

**Pull request merged**